Exiled.API/Extensions/SpawnExtensions.cs

```diff
@@ -0,0 +1,140 @@
+        /// </summary>
+        /// <param name="spawnLocation">The <see cref="SpawnLocation"/>.</param>
+        /// <returns>Returns the door name.</returns>
+        public static string GetDoorName(this SpawnLocation spawnLocation)
+        {
+            switch (spawnLocation)
+            {
+                case SpawnLocation.Inside012:
+                    return "012";
+                case SpawnLocation.Inside096:
+                    return "096";
+                case SpawnLocation.Inside914:
+                    return "914";
+                case SpawnLocation.InsideHid:
+                    return "HID";
+                case SpawnLocation.InsideGr18:
+                    return "GR18";
+                case SpawnLocation.InsideGateA:
+                    return "GATE_A";
+                case SpawnLocation.InsideGateB:
+                    return "GATE_B";
+                case SpawnLocation.InsideLczWc:
+                    return "LCZ_WC";
+                case SpawnLocation.InsideHidLeft:
+                    return "HID_LEFT";
+                case SpawnLocation.InsideLczCafe:
+                    return "LCZ_CAFE";
+                case SpawnLocation.Inside173Gate:
+                    return "173_GATE";
+                case SpawnLocation.InsideIntercom:
+                    return "INTERCOM";
+                case SpawnLocation.InsideHidRight:
+                    return "HID_RIGHT";
+                case SpawnLocation.Inside079First:
+                    return "079_FIRST";
+                case SpawnLocation.Inside012Bottom:
+                    return "012_BOTTOM";
+                case SpawnLocation.Inside012Locker:
+                    return "012_LOCKER";
+                case SpawnLocation.Inside049Armory:
+                    return "049_ARMORY";
+                case SpawnLocation.Inside173Armory:
+                    return "173_ARMORY";
+                case SpawnLocation.Inside173Bottom:
+                    return "173_BOTTOM";
+                case SpawnLocation.InsideLczArmory:
+                    return "LCZ_ARMORY";
+                case SpawnLocation.InsideHczArmory:
+                    return "HCZ_ARMORY";
+                case SpawnLocation.InsideNukeArmory:
+                    return "NUKE_ARMORY";
+                case SpawnLocation.InsideSurfaceNuke:
+                    return "SURFACE_NUKE";
+                case SpawnLocation.Inside079Secondary:
+                    return "079_SECOND";
+                case SpawnLocation.Inside173Connector:
+                    return "173_CONNECTOR";
+                case SpawnLocation.InsideServersBottom:
+                    return "SERVERS_BOTTOM";
+                case SpawnLocation.InsideEscapePrimary:
+                    return "ESCAPE_PRIMARY";
+                case SpawnLocation.InsideEscapeSecondary:
+                    return "ESCAPE_SECONDARY";
+                default:
+                    return default;
+            }
```

Comment:
Any reason to not just use an array?

Reply:
Not sure how an array would be useful for converting enums to base-game room names?

Reply:
I'm marking this resolved as it isn't much of an issue.